marine.c

```diff
@@ -362,39 +362,28 @@ marine_write_specified_fields(packet_filter *filter, epan_dissect_t *edt, char *
 
     proto_tree_children_foreach(edt->tree, proto_tree_get_node_field_values, &data);
 
-
-    //char *output = (char *) g_malloc0(4096); // todo this can overflow
-    int counter = 0;
     for (i = 0; i < fields->fields->len; ++i) {
-        if (0 != i) {
-            output[counter++] = fields->separator;
-        }
         if (NULL != fields->field_values[i]) {
             GPtrArray *fv_p;
             gchar *str;
             gsize j;
             fv_p = fields->field_values[i];
-            if (fields->quote != '\0') {
-                output[counter++] = fields->quote;
-            }
+            output[i] = (char *) g_malloc0(4096);
+            int counter = 0;
 
-            /* Output the array of (partial) field values */
             for (j = 0; j < g_ptr_array_len(fv_p); j++) {
                 str = (gchar *) g_ptr_array_index(fv_p, j);
                 for (char *p = str; *p != '\0'; p++) {
```

Comment:
At least limit this loop to 4095 iterations or something - better slicing the output than overflowing.
Have you tried figuring out the length properly? How does wireshark/tshark do it?

Reply:
I agree with @yehonatanz. If you can't figure out how Wireshark does it, `realloc` or limiting the loop should suffice.